src/main/java/bio/terra/service/dataset/flight/create/CreateDatasetGetOrCreateStorageAccountStep.java

```diff
@@ -40,11 +43,16 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
         workingMap.get(ProfileMapKeys.PROFILE_MODEL, BillingProfileModel.class);
     UUID datasetId = workingMap.get(DatasetWorkingMapKeys.DATASET_ID, UUID.class);
 
-    AzureStorageAccountResource storageAccount =
-        resourceService.getOrCreateDatasetStorageAccount(
-            DatasetJsonConversion.datasetRequestToDataset(datasetRequestModel, datasetId),
-            profileModel,
-            context.getFlightId());
+    AzureStorageAccountResource storageAccount;
+    try {
+      storageAccount =
+          resourceService.getOrCreateDatasetStorageAccount(
+              DatasetJsonConversion.datasetRequestToDataset(datasetRequestModel, datasetId),
+              profileModel,
+              context.getFlightId());
+    } catch (TransientDataAccessException | TransactionSystemException e) {
+      return new StepResult(StepStatus.STEP_RESULT_FAILURE_RETRY, e);
+    }
```

Comment:
If there were a different exception thrown, could `storageAccount` still be unitialized?

Reply:
If a different exception was thrown, it wouldn't be caught by the `catch`, so the method would exit with the thrown exception.

Either the `getOrCreateDatasetStorageAccount `completes successfully and `storageAccount` has a value, or an exception was thrown and we catch and return early or allow the exception to be thrown.